core/collector.class.inc.php

```diff
@@ -137,6 +140,47 @@ public function AttributeIsOptional($sAttCode)
 		return false; // By default no attribute is optional
 	}
 
+	/**
+	 * Determine if a given attribute null value is allowed to be transformed between collect and data synchro steps.
+	 *
+	 *  If transformed, its null value is replaced by '<NULL>' and sent to iTop data synchro.
+	 *  It means that existing value on iTop side will be kept as is.
+	 *
+	 *  Otherwise if not transformed, empty string value will be sent to datasynchro which means resetting current value on iTop side.
+	 *
+	 * Best practice:
+	 * for fields like decimal, integer or enum, we recommend to configure transformation as resetting will fail on iTop side (Bug N°776).
+	 *
+	 * The implementation is based on a predefined configuration parameter named from the
+	 * class of the collector (all lowercase) with _nullified_attributes appended.
+	 *
+	 * Example: here is the configuration to "nullify" the attribute 'location_id' for the class MyCollector:
+	 * <mycollector_nullified_attributes type="array">
+	 *    <attribute>location_id</attribute>
+	 * </mycollector_nullified_attributes>
+	 *
+	 * @param string $sAttCode
+	 *
+	 * @return boolean True if the attribute can be skipped, false otherwise
+	 */
+	public function AttributeIsNullified($sAttCode) {
+		if ($this->aNullifiedAttributes === null) {
+			$this->aNullifiedAttributes = Utils::GetConfigurationValue(get_class($this)."_nullified_attributes", null);
+			if ($this->aNullifiedAttributes === null) {
+				// Try all lowercase
+				$this->aNullifiedAttributes = Utils::GetConfigurationValue(strtolower(get_class($this))."_nullified_attributes", null);
+			}
+		}
```

Comment:
@odain-cbd Shouldn't this part be done in the class constructor? Now it will run this code for every field for every row if there aren't any nullified attributes..

Reply:
that's right. i had some doubt as well... 

I had the choice between doing like this and code will be executed further times for every field if there is no such configuration BUT only during collecting step. or moving it inside constructor and it will be executed once even for other steps where it does not make sense... 
maybe we could adapt that code to follow your advice (moving it into constructor) and checking which step will be executed (collecting or not).

FYI  method AttributeIsOptional works also like AttributeIsNullified and could be moved inside constructor as well.

Reply:
Okay, I have updated the code.